src/app/organization/offices/edit-office/edit-office.component.ts

```diff
@@ -0,0 +1,81 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+/** Custom Services */
+import { OrganizationService } from 'app/organization/organization.service';
+
+/**
+ * Edit Office component.
+ */
+@Component({
+  selector: 'mifosx-edit-office',
+  templateUrl: './edit-office.component.html',
+  styleUrls: ['./edit-office.component.scss']
+})
+export class EditOfficeComponent implements OnInit {
+
+  /** Selected Data. */
+  officeData: any;
+  /** Office form. */
+  officeForm: FormGroup;
+  /** Date. */
+  openedOn: any;
+
+    /**
+     * Retrieves the charge data from `resolve`.
+     * @param {ProductsService} organizationService Organization Service.
+     * @param {FormBuilder} formBuilder Form Builder.
+     * @param {ActivatedRoute} route Activated Route.
+     * @param {Router} router Router for navigation.
+     * @param {MatDialog} dialog Dialog reference.
+     * @param {DatePipe} datepipe Convert Date.
+     */
+    constructor(private organizationService: OrganizationService,
+      private formBuilder: FormBuilder,
+      private route: ActivatedRoute,
+      private router: Router,
+      private datepipe: DatePipe
+    ) {
+      this.route.data.subscribe((data: { officeTemplate: any }) => {
+        this.officeData = data.officeTemplate;
+        this.openedOn = `${this.officeData.openingDate[0]}-${this.officeData.openingDate[1]}-${this.officeData.openingDate[2]}`;
+      });
+    }
+
+  ngOnInit() {
+    this.createOfficeForm();
+  }
+
+  /**
+   * Create Edit Office Form.
+   */
+  createOfficeForm() {
+    this.officeForm = this.formBuilder.group({
+      'office': [this.officeData.name, Validators.required],
+      'parentOffice': [this.officeData.parentName, Validators.required],
+      'openedOn': [this.openedOn, Validators.required],
+      'externalId': [this.officeData.externalId],
+    });
+  }
+
+  /**
+   * Submit the edit office form and edit office.
+   *
+   */
+  submit() {
+    let formObj: any;
+    const name = this.officeForm.get('office').value;
+    const parentOffice = this.officeForm.get('parentOffice').value;
+    let openedOn = this.officeForm.get('openedOn').value;
+    openedOn = this.datepipe.transform(openedOn, 'dd MMMM yyyy');
+    const externalId = this.officeForm.get('externalId').value;
+    formObj = {name: name, parentId: this.officeData.parentId, locale: 'en', dateFormat: 'dd MMMM yyyy', openingDate: openedOn, externalId: externalId};
```

Comment:
Instead of using formObj variable, use const office = this.officeform.value, transform the opendon date and then use officeform.patchvalue to replace it, see how its done in #525 submit function.
```suggestion
```